Controllers/BookStoreItemController.cs

```diff
@@ -0,0 +1,187 @@
+        public IActionResult Create(string Product, [Bind("Available,Booked,Sold")] BookStoreItemDto bookStoreItem)
+        {
+            var bookedCount = _bookStoreService.Products()
+                .Where(b => b.Name == Product)
+                .FirstOrDefault().Bookings
+                .Where(s => s.StatusId == (int)Statuses.APPROVED || s.StatusId == (int)Statuses.SUBMITED).Count();
+
+            var soldCount = _bookStoreService.Products()
+                .Where(b => b.Name == Product)
+                .FirstOrDefault().Bookings
+                .Where(s => s.StatusId == (int)Statuses.COMPLETED).Count();
+
+            bookStoreItem.ProductId = _bookStoreService.Products().Where(p => p.Name == Product).FirstOrDefault().Id;
+            bookStoreItem.Booked = bookedCount;
+            bookStoreItem.Sold = soldCount;
+            bookStoreItem.Available = bookStoreItem.Available - (bookStoreItem.Booked + bookStoreItem.Sold);
+
+            if (ModelState.IsValid)
+            {
+                var storeMap = _mapper.Map<BookStoreItem>(bookStoreItem);
+                if (!_bookStoreService.Post(storeMap))
+                {
+                    ModelState.AddModelError("", "Something went wrong, store item was not saved");
+                    return StatusCode(500, ModelState);
+                }
+                return RedirectToAction(nameof(Index));
+            }
+            ViewData["Name"] = new SelectList(_bookStoreService.Products(), "Name", "Name");
+            return View(bookStoreItem);
+        }
+
+        // GET: BookStoreItem/Edit/5
+        public IActionResult Edit(int id)
+        {
+            if (id == null || _bookStoreService.Get(id) == null)
+            {
+                return NotFound();
+            }
+            _bookStoreService.Products().SelectMany(p => p.Bookings).Where(b => b.StatusId == (int)Statuses.SUBMITED || b.StatusId == (int)Statuses.APPROVED);
+            var bookStoreItem = _bookStoreService.Get(id);
+
+            if (bookStoreItem == null)
+            {
+                return NotFound();
+            }
+            ViewData["ProductId"] = bookStoreItem.Product.Id;
+            ViewData["ProductName"] = bookStoreItem.Product.Name;
+            ViewData["ProductBooked"] = bookStoreItem.Product.Bookings.Where(b => b.StatusId == (int)Statuses.SUBMITED || b.StatusId == (int)Statuses.APPROVED).Count();
+            ViewData["ProductSold"] = bookStoreItem.Product.Bookings.Where(b => b.StatusId == (int)Statuses.COMPLETED).Count();
+            return View(bookStoreItem);
+        }
+
+        // POST: BookStoreItem/Edit/5
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Edit(int id, [Bind("Id,Available,Booked,Sold,ProductId")] BookStoreItemDto bookStoreItemUpdate)
+        {
+            bookStoreItemUpdate.Available = bookStoreItemUpdate.Available - (bookStoreItemUpdate.Booked + bookStoreItemUpdate.Sold);
+            if (ModelState.IsValid)
```

Comment:
it's hard to read this block: extract validation to the separate method. Please do it for the other controllers